addons/mail/static/src/components/composer_suggested_recipient/composer_suggested_recipient.js

```diff
@@ -138,13 +138,28 @@ class ComposerSuggestedRecipient extends Component {
 
     /**
      * @private
+     * @param {object} record the newly-created record
      */
-    _onDialogSaved() {
+    _onDialogSaved(record) {
         const thread = this.suggestedRecipientInfo && this.suggestedRecipientInfo.thread;
         if (!thread) {
             return;
         }
         thread.fetchAndUpdateSuggestedRecipients();
+        if (!this.suggestedRecipientInfo.partner) {
+            this.env.services.notification.notify({
+                title: this.env._t('Invalid Partner'),
+                message: this.env._t('The information you have entered does not match the existing contact information for this record.\
+                             The partner was not created.'),
+                type: 'warning'
+            });
+            this.env.session.rpc('/web/dataset/call_kw/res.partner/unlink', {
```

Comment:
Prefer `this.env.services.rpc()` rather than `this.env.session.rpc()`

Reply:
Doing `rpc` from `session` is legacy way. Not entirely sure it's exactly the same behaviour, so better ensure to use the one from `env.services` (newer)